pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix

```
@@ -1,14 +1,20 @@
 { stdenv, lib, fetchurl, autoPatchelfHook, dpkg, awscli, unzip }:
 let
   ver = "1.2.331.0";
-  source =
-    if stdenv.isDarwin then {
-      url = "https://s3.amazonaws.com/session-manager-downloads/plugin/${ver}/mac/sessionmanager-bundle.zip";
-      sha256 = "0gr6frdn9jvxnkymkcpvgkqw4z2sac9jdf5qj4hzakq1zkfviazf";
-    } else {
-      url = "https://s3.amazonaws.com/session-manager-downloads/plugin/${ver}/ubuntu_64bit/session-manager-plugin.deb";
-      sha256 = "sha256-xWnY89dslkGhRTh9llRFkuUqYIjHQNt+TLnkPQr3u1Q=";
-    };
+  source = {
+    url = rec {
+      "x86_64-linux" = "https://s3.amazonaws.com/session-manager-downloads/plugin/${ver}/ubuntu_64bit/session-manager-plugin.deb";
+      "aarch64-linux" = "https://s3.amazonaws.com/session-manager-downloads/plugin/${ver}/ubuntu_arm64/session-manager-plugin.deb";
+      "x86_64-darwin" = "https://s3.amazonaws.com/session-manager-downloads/plugin/${ver}/mac/sessionmanager-bundle.zip";
+      "aarch64-darwin" = x86_64-darwin;
+    }."${system}";
+    sha256 = rec {
+      "x86_64-linux" = "sha256-NTkb6E0GRofd63TozLPxyNZ6GxO4pAc+MWsUDKWLZDY=";
+      "aarch64-linux" = "sha256-eB1Izvf/E0iOGkWt6fSc4oAt1sJ7XTxn/iwDNYz7WkA=";
+      "x86_64-darwin" = "sha256-TAphdcbzbNUZpWYVaHBkhI+RsgxWmI8n8Z6VwYxHXWg=";
+      "aarch64-darwin" = x86_64-darwin;
+    }."${system}";
```

```suggestion
    }.$system or (throw "Unsupported system: ${system}");
```
forgot to mention that. Same for line 10. Also I am not sure if that works, yet. system seems to be undefined.

---

It's not: [`error: undefined variable 'system'`](git@github.com:pimutils/khal.git).

`arch` isn't enough here, it also depends on what OS we're building for.

---

I think you want to replace ``system`` with ``stdenv.hostPlatform.system``.